api/middleware/apd.js

```diff
@@ -0,0 +1,108 @@
+const logger = require('../logger')('apd middleware');
+const {
+  apd: ApdModel,
+  apdActivity: defaultActivityModel
+} = require('../db').models;
+
+const cache = {};
+
+module.exports.loadApd = (model = ApdModel, idParam = 'id') => {
+  if (!cache[['loadApd', model, idParam]]) {
```

Comment:
This cacheing isn't super necessary, but it helps make our tests more robust.  It seems like a pretty low cost, but I'm open to discussion about it.